src/video.cpp

```diff
@@ -1281,6 +1281,10 @@ void encode_run(
   const encoder_t &encoder,
   void *channel_data) {
 
+  // set minimum frame time
+  auto minimum_frame_time = std::chrono::milliseconds(1000 / config::video.min_fps_target);
```

Comment:
I recommend putting a limit here to avoid violating the client-requested target framerate:
```
auto minimum_frame_time = std::chrono::milliseconds(1000 / std::min(config.framerate, config::video.min_fps_target));
```

Setting your new config to, e.g. 240, will cause the encoder to idle at >70fps when the stream is supposed to be 60fps. The problem with my suggested change, however, is that the idle framerate will only reach ~40fps, so the target framerate calculation is not quite accurate (as it should be closer to 60).

Reply:
I'm not sure why the calculation is off. Setting it to 20 does seem to give about 20 fps at idle, but above that and it doesn't seem linear. The config setting needs to be really high (~220) to achieve 60 fps.

Reply:
I changed the calculation to use your suggestion.

As of now, I'm not sure how to make it more accurate. Perhaps the range should just be 10-30 instead of opening it up so far.